### DISCUSSION TITLE
name prefixed routes not being filtered

### DISCUSSION THREAD

**Discussion by Miosame** [tighten/ziggy#649] [Q&A] [answered]
### Ziggy version

v1.5.0

### Laravel version

v10.7.1

### Description

```php
Route::middleware('auth')->group(function () {
    Route::name('hidden.')->group(function () {
        Route::get('/123', function () {

        })->name('whatever');
    });
});
```

```php
return [
    'except' => ['hidden.*'],
];
```

now ziggy will leak the route even though it should be prefixed as hidden.whatever in laravels routing
using it via inertia if that helps

**Comment by bakerkretzmar** [comment]
With the code you provided I can't reproduce this (route filtering works fine).

Please share exactly how you're using Ziggy (`@routes`? Vue plugin? Artisan command?) including your code, exactly what behaviour you're seeing (Ziggy includes the filtered route? Where exactly?), and what you expect instead.

**Comment by Miosame** [comment]
I was inspecting `(new Ziggy)->toArray()` inside `HandleInertiaRequests.php`'s share() method, if I was naming it `hidden.whatever` it would not be present, if I did the nested name prefix, it would show up and expose to the frontend.

I expected it to be equivalent in both cases, a name prefix for a group of routes and naming it hidden.whatever.

**Reply by bakerkretzmar** [reply, marked as answer]
It is equivalent, both those scenarios will result in one route called `hidden.whatever` and Ziggy will be able to filter on it. Try running `php artisan route:list` to see what your actual route names are in both cases. If you share more complete code examples I can try to help figure out what's going on in your app.

**Reply by Miosame** [reply]
as it turns out, deep in the codebase there was another same ->name()'d route that was overwriting this route, resulting in no prefix, the route:list was exactly what brought me there, thanks! hope this saves the headache to someone in the future coming across this :)
